demo/src/pages/when.astro

```diff
@@ -1,34 +1,33 @@
 ---
-import Document from '../layouts/Document.astro'
-import { When } from '@astropub/flow'
+import Document from "../layouts/Document.astro";
+import { When } from "@astropub/flow";
 ---
+
 <Document title="Astro When Component">
 	<h1>
 		The <code>&lt;When&gt;</code> Component
 	</h1>
 	<hr />
-	<pre><code
->&lt;When &#x7B;true&#x7D;&gt;
+	<pre><code>&lt;When &#x7B;true&#x7D;&gt;
 	&lt;h2&gt;True&lt;/h2&gt;
 	&lt;Fragment slot=&quot;else&quot;&gt;
 		&lt;h2&gt;False&lt;/h2&gt;
 	&lt;/Fragment&gt;
 &lt;/When&gt;</code></pre>
-	<When {true}>
+	<When condition={true}>
```

Comment:
Looking at the updated `when.astro`, I’m curious if this change is necessary. Should `<When>` require a specific key, or would it still evaluate all possible keys?

Reply:
It does not require a specific key but the key has to be a string. The typing is 
```ts
Record<string, unknown>
```

I think forcing the developer to be explicit is a good thing

Reply:
First, I agree with the stronger typing. Know that the shorthand `{true}` becomes `true={true}`, meaning it still works even with the `Record<string, unknown>` typing.

At the time of this writing, there is a bug in the VSCode extension that prevents Astro from recognizing any shorthand properties. I would rather not cater to that bug, since it is not actually a bug in the product itself.